### PR TITLE
TypeScript strict fixes

### PR DESCRIPTION
**Pull request by gsouquet**
- Strict fix for WhoIsTyping.ts
- Strict fix for VoipUserMapper.ts
- Strict fix for verifications.ts
- Strict fix for UserActivity.ts

## Checklist

-   [ ] (N/A) Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/UserActivity.ts [hunk 1]
```diff
@@ -182,17 +182,20 @@ export default class UserActivity {
...
     };
 
-    private onUserActivity = (event: MouseEvent): void => {
+    private onUserActivity = (event: Event): void => {
         // ignore anything if the window isn't focused
         if (!this.document.hasFocus()) return;
 
-        if (event.screenX && event.type === "mousemove") {
-            if (event.screenX === this.lastScreenX && event.screenY === this.lastScreenY) {
+        if (event.type === "mousemove" && (event as MouseEvent).screenX) {
```
Comment (t3chguy) [inline src/UserActivity.ts:189]: Can we use a custom type guard here instead? 

```
function isMouseEvent(event: Event): event is MouseEvent {
    return event.type.startsWith("mouse");
}
```

then using that before the `event.type` check to tell TS more about the type. Alternative an `isMouseMove` to be able to skip the event.type check

Comment (gsouquet) [reply src/UserActivity.ts]: Nice! TIL about type predicates

**Review by richvdh — CHANGES_REQUESTED** [review]
give me a ping when this passes CI ;)

File: src/VoipUserMapper.ts [hunk 2]
```diff
@@ -121,8 +121,12 @@ export default class VoipUserMapper {
         if (!LegacyCallHandler.instance.getSupportsVirtualRooms()) return;
 
         const inviterId = invitedRoom.getDMInviter();
+        if (!inviterId) {
+            logger.error("Could not find DM inviter for room id: " + invitedRoom.roomId);
```
Comment (richvdh) [inline src/VoipUserMapper.ts:125]: should this maybe `return` ?

**Review by richvdh — APPROVED** [review]
lgtm otherwise

**Pull request merged**